src/styles/themes/ThemeProvider.tsx

```diff
@@ -5,19 +5,24 @@ import CONST from '@src/CONST';
 import darkTheme from './default';
 import lightTheme from './light';
 import ThemeContext from './ThemeContext';
+import {ThemePreferenceWithoutSystem} from './types';
 import useThemePreference from './useThemePreference';
 
 const propTypes = {
     /** Rendered child component */
     children: PropTypes.node.isRequired,
 };
 
-function ThemeProvider(props: React.PropsWithChildren) {
+type ThemeProviderProps = React.PropsWithChildren & {
+    theme?: ThemePreferenceWithoutSystem;
+};
+
+function ThemeProvider({children, theme: themePreferenceProp}: ThemeProviderProps) {
     const themePreference = useThemePreference();
 
-    const theme = useMemo(() => (themePreference === CONST.THEME.LIGHT ? lightTheme : darkTheme), [themePreference]);
+    const theme = useMemo(() => ((themePreferenceProp ?? themePreference) === CONST.THEME.LIGHT ? lightTheme : darkTheme), [themePreference, themePreferenceProp]);
```

Comment:
this line is kind of hard to read, can we use a variable name to make it more clear what's happening here?